InnerEye/ML/deep_learning_config.py

```diff
@@ -317,20 +285,34 @@ def inference_options(self) -> Dict[ModelProcessing, Dict[ModelExecutionMode, Op
                 ModelExecutionMode.VAL: self.ensemble_inference_on_val_set,
             }
         }
-
-    def inference_on_set(self, model_proc: ModelProcessing, data_split: ModelExecutionMode) -> bool:
-        """
-        Returns True if inference is required for this model_proc and data_split.
-
-        :param model_proc: Whether we are testing an ensemble or single model.
-        :param data_split: Indicates which of the 3 sets (training, test, or validation) is being processed.
-        :return: True if inference required.
-        """
-        inference_option = self.inference_options()[model_proc][data_split]
+        inference_option = settings[model_proc][data_split]
         if inference_option is not None:
             return inference_option
 
-        return self.inference_defaults(model_proc, data_split)
+        # Defaults for when to run inference in the absence of any command line switches.
+        # This depends on ModelProcessing, perform_cross_validation, and ModelExecutionMode.
+        # If the current combination of these three parameters is not in this data structure,
+        # then default to False.
+        defaults: Dict[ModelProcessing, Dict[bool, Dict[ModelExecutionMode, bool]]] = {
+            ModelProcessing.DEFAULT: {
+                False: {
+                    ModelExecutionMode.TRAIN: False,
+                    ModelExecutionMode.VAL: False,
+                    ModelExecutionMode.TEST: True,
+                }
+            },
+            ModelProcessing.ENSEMBLE_CREATION: {
+                True: {
+                    ModelExecutionMode.TRAIN: False,
+                    ModelExecutionMode.VAL: False,
+                    ModelExecutionMode.TEST: True,
+                }
```

Comment:
False and True branches are now the same, is this correct?

Reply:
yes, I'd leave that in place though to make it more visible and easier to change if needed.

Reply:
(yes, I have deliberately turned off validation set inference by default)